app/models/data_set.rb

```diff
@@ -65,8 +65,12 @@ def places_for_postcode(postcode, distance = nil, limit = nil)
       end
     else
       location_data = MapitApi.location_for_postcode(postcode)
-      location = Point.new(latitude: location_data.lat, longitude: location_data.lon)
-      places_near(location, distance, limit)
+      if location_data.lat && location_data.lon
```

Comment:
I'm wondering whether it would read better if this logic was moved to the `places` method, with a guard clause like:
`return [] unless location.latitude && location.longitude`

Just a thought, happy to be told otherwise